sympy/physics/mechanics/joint.py

```diff
@@ -44,18 +45,48 @@ class Joint(ABC):
         Generalized coordinates of the joint.
     speeds : List of dynamicsymbols, optional
         Generalized speeds of joint.
-    parent_joint_pos : Vector, optional
-        Vector from the parent body's mass center to the point where the parent
-        and child are connected. The default value is the zero vector.
-    child_joint_pos : Vector, optional
-        Vector from the child body's mass center to the point where the parent
-        and child are connected. The default value is the zero vector.
+    parent_point : Point or Vector, optional
+        Attachment point where the joint is fixed to the parent body. If a
+        vector is provided, than the attachment point is computed by adding the
+        vector to the body's mass center. The default value is the parent's mass
+        center.
+    child_point : Point or Vector, optional
+        Attachment point where the joint is fixed to the child body. If a
+        vector is provided, than the attachment point is computed by adding the
+        vector to the body's mass center. The default value is the child's mass
+        center.
     parent_axis : Vector, optional
-        Axis fixed in the parent body which aligns with an axis fixed in the
-        child body. The default is x axis in parent's reference frame.
+        .. deprecated:: 1.12
+            Axis fixed in the parent body which aligns with an axis fixed in the
+            child body. The default is the x axis of parent's reference frame.
+            For more information on this deprecation, see
+            :ref:`deprecated-mechanics-joint-axis`.
     child_axis : Vector, optional
-        Axis fixed in the child body which aligns with an axis fixed in the
-        parent body. The default is x axis in child's reference frame.
+        .. deprecated:: 1.12
+            Axis fixed in the child body which aligns with an axis fixed in the
+            parent body. The default is the x axis of child's reference frame.
+            For more information on this deprecation, see
+            :ref:`deprecated-mechanics-joint-axis`.
+    parent_interframe : ReferenceFrame, optional
+        Interframe of the joint fixed to the parent body, i.e. frame in the of
+        parent body with respect to which the joint is formulated.  If a Vector
```

Comment:
This first sentence isn't clear, possible typo?

Reply:
It should have been: `Interframe of the joint fixed to the parent body, i.e. frame in the parent body with respect to which the joint is formulated.`
However I think it could be made clearer by using one of the following:
1. Joint attachment frame of the parent body, i.e. frame fixed to the parent body with respect to which the joint transformation is formulated.
2. Intermediate frame of the parent body with respect to which the joint transformation is formulated.

Which one do you think is more clear? Or do you have a better suggestion?

Reply:
I think #2 is clearest.